### ISSUE TITLE
Pagination items not read out with mouse tracking in NVDA and Chrome

### ISSUE THREAD

**Issue by selfthinker** [alphagov/govuk-frontend#5977]
## Description of the issue

Nothing at all is read out when hovering over items of the [pagination component](https://design-system.service.gov.uk/components/pagination/) with a mouse in NVDA and Chrome with mouse tracking enabled.

## Steps to reproduce the issue

1. Use NVDA with mouse tracking enabled (via Preferences > Settings... > Mouse > Enable mouse tracking)
2. Open [an example page](https://design-system.service.gov.uk/components/pagination/for-list-pages/) in Chrome
3. Hover over the page numbers or 'Previous'/'Next'

## Actual vs expected behaviour

Nothing is read out except the ellipsis.
It should read out the numbers and 'Previous'/'Next'.

The items all have an `aria-label`, for example, the link with the number 1 has one that says "Page 1". While most other screen readers would read the ARIA label, [NVDA intentionally does not](https://github.com/nvaccess/nvda/issues/5600). NVDA only reads what is visible on the screen when navigating via hovering with the mouse.
Therefore, it is fine that it doesn't read "Page 1", but it should read "1".

## Environment

This only happens with NVDA in Chrome.
It doesn't happen with NVDA in Firefox.
And it doesn't happen with any of the other desktop screen readers we usually test in (JAWS, VO, Narrator).

- GOV.UK Frontend Version: latest
- Operating system: Windows 10 Pro
- Browser: Chrome
- Browser version: 132
- Screen reader: NVDA
- Screen reader version: 2024.4

I assume this is either an issue in NVDA or Chrome.
I also tried an old version of NVDA (2018.1) with an old version of Chrome (64).
The issue didn't happen in that combination.

## Further findings

I've created a few [reduced test cases](https://jsbin.com/baragux) and found that the reason why this is happening is the method / hack that makes more than just the link clickable. That method uses CSS-generated content / a pseudo element that sort of goes on top of the link.

```css
.govuk-pagination__link:after {
    content: "";
    position: absolute;
    top: 0;
    right: 0;
    bottom: 0;
    left: 0;
}
```

My assumption is that NVDA doesn't "see" the text because the pseudo element is on top of it.

There is a seemingly [related issue](https://github.com/alphagov/govuk-frontend/issues/2278) where the same thing happens with checkboxes. But I don't believe it is directly related as native checkboxes without any CSS are also affected while a native list of links is not.

**Comment by selfthinker** [comment]
> I wonder if we should generally test all of our components with mouse tracking

I've just created a small story for that purpose: https://github.com/alphagov/govuk-frontend/issues/5978
